### DISCUSSION TITLE
Automating Dependency Updates

### DISCUSSION THREAD

**Discussion by kavithachandramohan-okta** [MobileNativeFoundation/discussions#171] [General]
Hi folks, 

We are looking into automating dependency updates and wanted to know the tools and process that has helped your teams.
I am looking into Github Dependabot and that looks promising. 
 Also how far have you taken the automation,  creating PR vs auto merging if tests pass?  

Looking forward to hearing about your experience, Thanks all

**Comment by keith** [comment]
We used to do auto-updates like you describe weekly with CocoaPods on iOS, but at some point when we changed around our tools we stopped doing that, and at this point I think for us to do auto-updates again we'd still want the owners to shepherd the changes in to reduce risk.

I would love to use dependabot for this for complex dependency trees, and that seems possible for android builds if your source of truth is a pom.xml file, but not for iOS builds.

**Comment by mshearer123** [comment]
How about [renovate](https://github.com/renovatebot/renovate), we're hoping to set it up soon.

**Reply by kavithachandramohan-okta** [reply]
That was another tool under consideration, it would be great to hear from someone using it.

**Comment by philipphofmann** [comment]
I think Dependabot doesn't work for CocoaPods, Swift Package Manager, or Carthage; see https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

For other dependencies such as bundler and GH actions, I like Dependabot, and in an open-source repo, we [run it weekly](https://github.com/getsentry/sentry-cocoa/blob/master/.github/dependabot.yml).
